### PR TITLE
[android] added wearable module

### PR DESCRIPTION
**Pull request by cammace**
Adds a wearable testapp module. 

cc: @zugaldia or @tobrun for review

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/activity/FeatureOverviewActivity.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+
+public class FeatureOverviewActivity extends WearableActivity implements FeatureAdapter.ItemSelectedListener {
+
+  private static final String TAG = "MainActivity";
+  private WearableRecyclerView wearableRecyclerView;
+  private ArrayList<Feature> exampleItemModels;
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/activity/FeatureOverviewActivity.java]: Is there are reason for not using List above ArrayList here?

File: platform/android/MapboxGLAndroidSDKWearTestApp/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <string name="app_name">MapboxGLAndroidSDKWearTestApp</string>
+
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKWearTestApp/src/main/res/values/strings.xml]: nit whiteline

File: platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/utils/OffsettingHelper.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  private float progressToCenter;
+
+  public OffsettingHelper() {
+  }
+
+  @Override
+  public void updateChild(View child, WearableRecyclerView parent) {
+    super.updateChild(child, parent);
+
+
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/utils/OffsettingHelper.java]: nit whiteline

File: platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/adapter/FeatureAdapter.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import android.widget.TextView;
+
+import com.mapbox.weartestapp.R;
+import com.mapbox.weartestapp.model.Feature;
+
+import java.util.ArrayList;
+
+public class FeatureAdapter extends WearableRecyclerView.Adapter<FeatureAdapter.ViewHolder> {
+
+  private ArrayList<Feature> data;
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/adapter/FeatureAdapter.java]: Is there are reason for not using List above ArrayList here?

File: platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/activity/FeatureOverviewActivity.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    exampleItemModels = new ArrayList<>();
+    exampleItemModels.add(new Feature(R.string.activity_simple_mapview_title, new Intent(FeatureOverviewActivity.this, SimpleMapViewActivity.class)));
+    //exampleItemModels.add("Show user location");
+
+    FeatureAdapter exampleAdapter = new FeatureAdapter(FeatureOverviewActivity.this, exampleItemModels);
+    wearableRecyclerView.setAdapter(exampleAdapter);
+
+    exampleAdapter.setListener(this);
+
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/activity/FeatureOverviewActivity.java]: nit whiteline

File: platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/activity/SimpleMapViewActivity.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap mapboxMap) {
+
+        // Customize map with markers, polylines, etc.
+      }
+    });
+
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKWearTestApp/src/main/java/com/mapbox/weartestapp/activity/SimpleMapViewActivity.java]: nit whiteline

File: platform/android/MapboxGLAndroidSDKWearTestApp/build.gradle [hunk 7]
```diff
@@ -0,0 +1,37 @@
+apply plugin: 'com.android.application'
+
+
+android {
+    compileSdkVersion 24
+    buildToolsVersion "24.0.2"
+
+    defaultConfig {
+        applicationId "com.mapbox.mapboxglandroidsdkweartestapp"
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKWearTestApp/build.gradle]: different applicationId?

File: platform/android/.gitignore [hunk 8]
```diff
@@ -21,6 +21,7 @@ local.properties
 
 # Token file
 MapboxGLAndroidSDKTestApp/src/main/res/values/developer-config.xml
+MapboxGLAndroidSDKWearTestApp/src/main/res/values/developer-config.xml
```
Comment (tobrun) [inline platform/android/.gitignore]: Is the gradle task responsible for generating `MapboxGLAndroidSDKTestApp/src/main/res/values/developer-config.xml` also capable of generating `MapboxGLAndroidSDKWearTestApp/src/main/res/values/developer-config.xml`?

**Comment by cammace** [comment]
Changed `ArrayList` to `List` and other changes @tobrun pointed out. I also added `checkstyle` to gradle file.

**Pull request merged**